source/dotnet/wholesale-api/CalculationResults/CalculationResults.UnitTests/Infrastructure/CalculationResultClient/TableRowsTests.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright 2020 Energinet DataHub A/S
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Energinet.DataHub.Wholesale.CalculationResults.Infrastructure.CalculationResultClient;
+using FluentAssertions;
+using Xunit;
+using Xunit.Categories;
+
+namespace Energinet.DataHub.Wholesale.CalculationResults.UnitTests.Infrastructure.CalculationResultClient;
+
+[UnitTest]
+public class TableRowsTests
+{
+    private readonly List<string> _columnNames;
+    private readonly List<string[]> _rowData;
+
+    public TableRowsTests()
+    {
+        _columnNames = new List<string>() { "col1", "col2", "col3", };
+        _rowData = new List<string[]>() { new[] { "cell11", "cell12", "cell13" }, new[] { "cell21", "cell22", "cell23" }, };
+    }
+
+    [Fact]
+    public void Count_ReturnsNumberOfRows()
+    {
+        // Arrange
+        var sut = new TableRows(_columnNames, _rowData);
+
+        // Act
+        var rowCount = sut.Count;
+
+        // Assert
+        rowCount.Should().Be(2);
+    }
+
+    [Fact]
+    public void IndexerWithRowIndex_ReturnsCorrectRow()
+    {
+        // Arrange
+        var expectedRow = new string[] { "cell21", "cell22", "cell23" };
+        var sut = new TableRows(_columnNames, _rowData);
+
+        // Act
+        var actualRow = sut[1];
+
+        // Assert
+        actualRow.Should().Equal(expectedRow);
+    }
+
+    [Fact]
+    public void IndexerWithColumnName_ReturnsCorrectCell()
+    {
+        // Arrange
+        var sut = new TableRows(_columnNames, _rowData);
+
+        // Act
+        var actual = sut[1, "col2"];
+
+        // Assert
+        actual.Should().Be("cell22");
+    }
+
+    [Fact]
+    public void IndexerWithInvalidColumnName_ThrowsException()
+    {
+        // Arrange
+        var sut = new TableRows(_columnNames, _rowData);
+
+        // Act + Assert
+        Assert.ThrowsAny<Exception>(() => sut[1, "invalid_column_name"]);
+    }
+
+    [Fact]
+    public void IndexerWithInvalidRowIndex_ThrowsException()
+    {
+        // Arrange
+        var sut = new TableRows(_columnNames, _rowData);
+
+        // Act + Assert
+        Assert.ThrowsAny<Exception>(() => sut[100, "col2"]);
```

Comment:
General idea: How about not making me check these values by being explicit? And then have the reusable valid/invalid values defined close to the other class level data.
```suggestion
        // Arrange
        var sut = new TableRows(_columnNames, _rowData);

        // Act + Assert
        Assert.ThrowsAny<Exception>(() => sut[_invalidRowIndex, _validColumnName]);
```